src/geometry/geom2/canonicalize.js

```diff
@@ -0,0 +1,24 @@
+const vec2 = require('../../math/vec2')
+
+/**
+ * Produces a canonicalized geometry by canonicalizing the sides.
+ * Must be called before exposing any side data.
```

Comment:
I think we should be more explicit about what canonicalizing means in the first place

Reply:
I'm not a fan of the term. As of NOW, the logic just 'applies' the tranforms, so I suggest we change to...

- canonicalize() ==> applyTransforms()
- isCanonicalized ==> isTransformed

Reply:
I agree that the naming is better ! 
so we are clear on things
- raw geometry
- transform 
- applyTransform => sets the actual transforms, sets the 'isTransformed' flag

What happens if you apply multiple transforms ?

Reply:
@z3dev did you see my question above ? :)

Reply:
Sorry.

See transforms(), as that function resets the isTransformed = false, and does the matrix multiplication.

In addition, it’s really important that the contents are accessed via toPolnts, toSide, or toPolygons. Those functions will apply the transforms and returns the contents.

Reply:
@kaosat-dev Nice catch. There were some inconsistencies in the way transforms were being handled, as well as the isCanonicalized state. These were fixed. However, I would like to add test cases to make sure that the 'lazy' application REALLY works.

Reply:
ah sorry, I had missed that !
but that makes sense, also good idea on tests

Reply:
about numeric transformations and precision issues: we still need to check what happens inside of booleans and how 'very close' points are merged into a single one, but perhaps that should be another PR as it is a booleans issue

Reply:
@kaosat-dev please review geom2 only.

i added a full scenario to test lazy transforms.

also, I found something very interesting. basically, the isCanonicalized boolean is unnecessary because the transforms are always reset to the indentity matrix. the transform() function only changes the transforms matrix, while the canonicalize() function applies the transforms then resets the transforms matrix.

Reply:
@kaosat-dev please review geom2 again. how do those changes look to transform() and canonicalize() ?